app/src/main/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -0,0 +1,39 @@
+package com.intive.picover.auth.repository
+
+import com.google.firebase.auth.FirebaseAuth
+import com.intive.picover.auth.model.AuthEvent
+import javax.inject.Inject
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.tasks.await
+
+class AuthRepository @Inject constructor(
+	private val firebaseAuth: FirebaseAuth,
+) {
+
+	fun observeEvents() =
+		callbackFlow {
+			val authStateListener = FirebaseAuth.AuthStateListener {
+				if (it.currentUser == null) {
+					AuthEvent.NotLogged
+				} else {
+					AuthEvent.Logged
+				}.let { authEvent ->
+					trySend(authEvent)
+				}
+			}
+			firebaseAuth.addAuthStateListener(authStateListener)
+			awaitClose { firebaseAuth.removeAuthStateListener(authStateListener) }
+		}
+
+	fun logout() {
+		firebaseAuth.signOut()
+	}
+
+	// TODO #100 Handle FirebaseAuthRecentLoginRequiredException
+	suspend fun deleteAccount() {
+		firebaseAuth.currentUser!!
```

Comment:
Of course, we can avoid it and use the safe call `?.` but it could lead to a hidden not working option (nothing will happen after clicking, but the app will not crash). Because of that I decided to force non-nullable here, and the app should crash when the user is trying to delete the account but Firebase does not have currentUser. Thanks to that we can track all edge-cases.